nasa/ow_simulator
# Added a dynamic reconfigure option to auto clear goal errors

## Linked Issues:
Jira Ticket 🎟️ | [OCEANWATER-1248](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-1248)

## Summary of Changes
* All world launch files have an option to automatically clear goal errors when an action in the goal error category completes successfully. 
  * This is now the default behavior, but the more complex behavior that requires the use of FaultClear can still be used by setting the flag to false at simulation launch or modifying during runtime via the ros parameter server. 
* Made a category in the faults section of Dynamic Reconfigure called "fault_behaviors" for both "arm_motion_continues_in_fault" and "auto_clear_goal_errors".

## Test Default (simple) Fault Behavior
1. Launch any simulation world 
2. Monitor faults `rostopic echo /system_fault_status/value`. It should start at 0.
3. Call an impossible action, e.g. `rosrun ow_lander arm_move_cartesian.py -x 100`. It will fail, and system_fault_status will be 2 or whatever the value of the action category you called. 
4.  Call a possible action in the same category, e.g. `rosrun ow_lander arm_move_cartesian.py -x 1.8 -z 0.5`. At the start of the action the system fault topic will go back to 0 and the action should complete.  
5. Go into RQT GUI, open Dynamic Reconfigure tab, go to faults, and under the "fault_behaviors" section untick "auto_clear_goal_errors". 
6. Call an impossible action, e.g. `rosrun ow_lander arm_move_cartesian.py -x 100`. It will fail, and system_fault_status will be 2 or whatever the value of the action category you called. 
7.  Call a possible action in the same category, e.g. `rosrun ow_lander arm_move_cartesian.py -x 1.8 -z 0.5`. The action  will be rejected. 
8. Clear the goal error `rosrun ow_lander fault_clear.py 2`, or use the value correct for the action category you used. The echo'd value should now be zero.
9.  Call a possible action in the same category, e.g. `rosrun ow_lander arm_move_cartesian.py -x 1.8 -z 0.5`. The action should  complete successfully. 

## Notes on FaultClear
I decided not to produce a warning/error when FaultClear is used and auto_clear_goal_errors is true because one could imagine a user's autonomy to be configured such that clearing goal errors is still necessary before issuing a follow-up lander action.
